keras_nlp/models/bert/bert_preprocessing.py

```diff
@@ -167,15 +168,27 @@ class BertPreprocessor(keras.layers.Layer):
 
     This preprocessing layer will do three things:
 
-     - Tokenize any number of inputs using the `tokenizer`.
+     - Tokenize any number of input segments using the `tokenizer`.
      - Pack the inputs together using a `keras_nlp.layers.MultiSegmentPacker`.
        with the appropriate `"[CLS]"`, `"[SEP]"` and `"[PAD]"` tokens.
      - Construct a dictionary of with keys `"token_ids"`, `"segment_ids"`,
        `"padding_mask"`, that can be passed directly to a BERT model.
 
-    This layer will accept either a tuple of (possibly batched) inputs, or a
-    single input tensor. If a single tensor is passed, it will be packed
-    equivalently to a tuple with a single element.
+    This layer can be used directly with `tf.data.Dataset.map` to preprocess
+    string data in the `(x, y, sample_weight)` format used by
+    `keras.Model.fit`.
+
+    The call method of this layer accepts three arguments, `x`, `y`, and
+    `sample_weights`. `x` should be either a (possible batched) string tensor,
+    or a tuple of (possible batched) string tensors. `y` and `sample_weights`
+    are both optional, can have any format, and will be passed through
+    unaltered.
+
+    Special care should be taken when using `tf.data` to map over an unlabeled
```

Comment:
This is also true with our current layer, it is because tf.data will attempt to unpack a tuple directly into the function passed. Not sure how to best document this gotcha. Could do an example if that would be better. Or both.